processor/k8sprocessor/kube/client.go

```diff
@@ -243,14 +295,51 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 	}
 
 	for _, r := range c.Rules.Labels {
-		if v, ok := pod.Labels[r.Key]; ok {
-			tags[r.Name] = c.extractField(v, r)
+		if r.From == "pod" {
+			if v, ok := pod.Labels[r.Key]; ok {
+				tags[r.Name] = c.extractField(v, r)
+			}
+		}
+	}
+
+	for _, r := range c.Rules.Annotations {
+		if r.From == "pod" {
+			if v, ok := pod.Annotations[r.Key]; ok {
+				tags[r.Name] = c.extractField(v, r)
+			}
+		}
+	}
+	return tags
+}
+
+func (c *WatchClient) extractNamespaceAttributes(namespace *api_v1.Namespace) map[string]string {
+	tags := map[string]string{}
+
+	if c.Rules.NamespaceUID {
+		uid := namespace.GetUID()
+		tags[tagNamespaceUID] = string(uid)
+	}
+
+	if c.Rules.NamespaceStartTime {
+		ts := namespace.GetCreationTimestamp()
+		if !ts.IsZero() {
+			tags[tagNamespaceStartTime] = ts.String()
+		}
+	}
+
+	for _, r := range c.Rules.Labels {
+		if r.From == "namespace" {
```

Comment:
I'd define the `From` field as a custom string type or at least use constants for the values (namespace, pod)